src/envelope/spacecharge/EnvelopeSpaceChargePush.cpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Marco Garten, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "EnvelopeSpaceChargePush.H"
+
+#include <AMReX_BLProfiler.H>
+#include <AMReX_REAL.H>       // for Real
+#include <AMReX_SmallMatrix.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Print.H>
+
+#include <cmath>
+
+namespace impactx::spacecharge
+{
+        void
+        envelope_space_charge2D_push (
+        [[maybe_unused]] RefPart const & refpart,
+        Map6x6 & cm,
+        amrex::ParticleReal current,
+        amrex::ParticleReal ds
+    )
+    {
+        using namespace amrex::literals;
+
+        // initialize the linear transport map
+        Map6x6 R = Map6x6::Identity();
+
+        // physical constants and reference quantities
+        amrex::ParticleReal const c = ablastr::constant::SI::c;
+        amrex::ParticleReal const ep0 = ablastr::constant::SI::ep0;
+        amrex::ParticleReal const pi = ablastr::constant::math::pi;
```

Comment:
I think you can also write:
```suggestion
        using ablastr::constant::SI::c;
        using ablastr::constant::SI::ep0;
        using ablastr::constant::math::pi;
```

Reply:
done :)